sdl/v2_test.go

```diff
@@ -192,3 +193,10 @@ func Test_v1_Parse_ProfileNameNotServiceName(t *testing.T) {
 	require.NoError(t, err)
 	assert.Len(t, mani.GetGroups(), 1)
 }
+
+func Test_v2_Parse_DeploymentNameServiceNameMismatch(t *testing.T) {
+	sdl, err := ReadFile("./_testdata/deployment-svc-mismatch.yaml")
+	require.Nil(t, sdl)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "no service profile named")
```

Comment:
The problem was that it was not registering endpoints, everything else worked fine.  It was looking for endpoints by compute profile name, not service name.

```yaml
services:
  web:
    # ...
profiles:
  compute:
    bew:
      # ...
```

Reply:
i probably missed point here. fix makes endpoints iteration via service name. if deployment points to non-existing service it fails on `ReadFile`

valid
```yaml
services:
  web:
    # ...
profiles:
  compute:
    bew:
      # ...
  placement:
    westcoast:
      # ...
deployment:
  web:
    westcoast:
      profile: bew
```

invalid
```yaml
services:
  web:
    # ...
profiles:
  compute:
    bew:
      # ...
  placement:
    westcoast:
      # ...
deployment:
  bew:
    westcoast:
      profile: bew
```

Reply:
It wasn't failing it just wasn't picking up the endpoints.

Reply:
and yeah, your first example would have triggered it.

Reply:
updated test